### PR TITLE
Recommend no requests to dedicated masters

### PR DESCRIPTION
**Pull request by DaveCTurner**
Today the docs on node roles say that you shouldn't use dedicated
masters for heavy requests such as indexing and searching, but as per
the "designing for resilience" docs this guidance applies to all client
requests. This commit generalises the node roles docs slightly to
clarify this.

Relates #70435

### CODE REVIEW

File: docs/reference/modules/node.asciidoc [hunk 1]
```diff
@@ -132,14 +132,14 @@ It is important for the health of the cluster that the elected master node has
 the resources it needs to fulfill its responsibilities. If the elected master
 node is overloaded with other tasks then the cluster may not operate well. In
 particular, indexing and searching your data can be very resource-intensive, so
-in large or high-throughput clusters it is a good idea to avoid using the
-master-eligible nodes for tasks such as indexing and searching. You can do this
-by configuring three of your nodes to be dedicated master-eligible nodes.
+in large or high-throughput clusters you should avoid using the master-eligible
+nodes for other tasks such as indexing and searching. You can do this by
```
Comment (henningandersen) [inline docs/reference/modules/node.asciidoc:136]: I think this needs a bit of rewording, perhaps replace the sentence from "In particular..." with:

"In large large clusters you should avoid using the master-eligible nodes for other tasks, in particular tasks such as indexing and searching that can be very resource-intensive.

**Review by henningandersen — APPROVED** [review]
LGTM, thanks David.

File: docs/reference/modules/node.asciidoc (hunk 1 above)
Comment (DaveCTurner) [reply docs/reference/modules/node.asciidoc]: Thanks Henning, yes, that could be better. I pondered for a while and decided I couldn't find a great way to keep the bit about indexing and searching, see be45ea42847. WDYT?

**Review by henningandersen — APPROVED** [review]
LGTM.

**Pull request merged**